src/network/websockets/server/index.ts

```diff
@@ -63,12 +79,16 @@ export function runWebsocketServer<AuthDetails>(params: Params<AuthDetails>) {
             close: () => socket.terminate()
           })
           .catch(() => undefined);
+        processQueue();
       },
       push: async msg => {
         const { kind, id, lastSeenRevision, value } = msg;
         if (auth) {
           const details = await authDetails;
```

Comment:
Realize this probably isn't directly related to this PR but it's kind of weird here how `authDetails` is a Promise and `details` is the actual details. This could be cleaned up by
1) defining `authDetails` like this:
```typescript
let authDetails: AuthDetails | undefined = undefined;
```
2) writing `handlers.auth` as `async`:
```typescript
auth: async msg => {
  const { token } = msg;
  if (!auth) throw new Error("Unexpected auth message");
  authDetails = await auth
    .parseToken(token, {
      close: () => socket.terminate()
    })
    .catch(() => undefined);
  processCache();
},
```
3) For `handlers.push()`, remove the `async` and change the `if` block:
```typescript
  if (auth) {
    if (!authDetails) {
      addToQueue(msg);
      return;
    }
    if (!auth.canWrite({ auth: authDetails, kind, id })) return;
  }
```
4) also do the same with the function that's passed as the last parameter to `getAndObserve()` in `handlers.subscribe()`:
```typescript
v => {
  if (auth) {
    if (!authDetails) {
```

Reply:
I was sure I’ve left a comment on this before, sorry for the delay.

You’re right in it’s confusing, you’re not right that we can rewrite as you suggest. The promise in authDetails is actually another hidden queue. Because this code is confusing I have made #44 to clarify what is happening, see if that makes more sense!